projects/demo/src/modules/components/input-number/examples/7/index.ts

```diff
@@ -0,0 +1,31 @@
+import {Component} from '@angular/core';
+import {changeDetection} from '@demo/emulate/change-detection';
+import {encapsulation} from '@demo/emulate/encapsulation';
+import {TUI_DEFAULT_NUMBER_FORMAT, TUI_NUMBER_FORMAT} from '@taiga-ui/core';
+import {TUI_LANGUAGE, TuiLanguage} from '@taiga-ui/i18n';
+import {BehaviorSubject, map, Observable, switchMap} from 'rxjs';
+
+@Component({
+    selector: 'tui-input-number-example-7',
+    templateUrl: './index.html',
+    encapsulation,
+    changeDetection,
+    providers: [
+        {
+            provide: TUI_NUMBER_FORMAT,
+            deps: [TUI_LANGUAGE],
```

Comment:
What do you guys think if we ban `deps` in factories in favor of `inject`? It is better typed. Maybe not ban, but at least prefer.

Reply:
Yes, I also prefer `inject` instead of `deps`. But real type of `TUI_LANGUAGE` in our demo is different than type of token due to [LanguageSwitcher](https://github.com/taiga-family/taiga-ui/blob/main/projects/i18n/switch/language-switcher.providers.ts#L14). So it is just to avoid ugly typecasting like `as unknown as Observable<Observable<TuiLanguage>>>`